### PR TITLE
[diem genesis] some simplifications in Genesis module

### PR DESCRIPTION
**Pull request by sblackshear**
- DiemVersion::initialize takes the initial version number as input
- Set Diem version in Move code instead of Rust

### CODE REVIEW

File: language/tools/vm-genesis/src/lib.rs [hunk 1]
```diff
@@ -256,21 +255,7 @@ fn create_and_initialize_main_accounts(
...
-        session,
-        log_context,
-        DIEM_VERSION_MODULE_NAME,
-        "set",
-        vec![],
-        serialize_values(&vec![
-            MoveValue::Signer(root_diem_root_address),
-            MoveValue::U64(
-                /* Diem Framework major version number */ DIEM_MAX_KNOWN_VERSION.major,
-            ),
```
Comment (davidiw) [inline language/tools/vm-genesis/src/lib.rs:273]: c'mon this is so glorious looking!

**Review by davidiw — APPROVED** [review]

**Review by tzakian — APPROVED** [review]
Nice simplification! LGTM

**Comment by sblackshear** [comment]
It looks like this is failing the SDK compatibility test (probably it changes the genesis state/corresponding root hash)? So I am not sure if we can easily make this change...

**Review by tzakian — DISMISSED** [review]

**Pull request merged**